### PR TITLE
error in tests on FutureWarning (+ workarounds for when we can't)

### PR DESCRIPTION
**Pull request by dchudz**
I think this addresses much of https://github.com/HypothesisWorks/hypothesis-python/issues/1124 by:

(Updated description):

- move the tox setenv/getenv to where it'll do something
- use `tox-venv` so that we can turn on DeprecationWarnings as errors in 3.6 (but still can't in 2.7)
- ignore a bunch of deprecation warnings we can't do anything about
- fix the one actual problem this found with our package

### CODE REVIEW

**Comment by Zac-HD** [comment]
LGTM, pending conversation in the linked issue.

**Comment by dchudz** [comment]
(Some of the environments still need fixing.)

**Comment by Zac-HD** [comment]
FWIW I'm fine with only checking all the future and deprecation warnings on the latest version of Python.

Also fine with ignoring warnings for Faker, as we'll be taking it out soon - see #1134.

File: tox.ini [hunk 1]
```diff
@@ -2,31 +2,44 @@
...
-  LC_ALL
-  COVERAGE_FILE
-
-setenv=
-  PYTHONWARNINGS={env:PYTHONWARNINGS:error::DeprecationWarning,error::FutureWarning}
-
 [testenv]
-deps =
-    -rrequirements/test.txt
-    benchmark: pytest-benchmark==3.0.0
```
Comment (dchudz) [inline tox.ini]: (`benchmarks` are a thing of the past, I think)

Comment (Zac-HD) [reply tox.ini]: They are - well spotted.  Looks like they were removed from the Makefile at the time but not tox.ini

**Comment by dchudz** [comment]
Closing as per https://github.com/HypothesisWorks/hypothesis-python/issues/1124#issuecomment-373216896.

https://github.com/HypothesisWorks/hypothesis-python/pull/1149/files replaces part of this.

**Pull request closed without merging**